Source/Particles/WarpXParticleContainer.cpp

```diff
@@ -295,13 +295,17 @@ WarpXParticleContainer::AddNParticles (int /*lev*/, long n,
         );
     }
 
+    // Move particles to their appropriate tiles
+    Redistribute();
+
     // Remove particles that are inside the embedded boundaries
 #ifdef AMREX_USE_EB
     auto & distance_to_eb = WarpX::GetInstance().GetDistanceToEB();
     scrapeParticles( *this, amrex::GetVecOfConstPtrs(distance_to_eb), ParticleBoundaryProcess::Absorb());
 #endif
 
-    Redistribute();
+    // Call (local) redistribute again to remove particles with invalid ids
+    Redistribute(0, -1, 0, 1, 1);
```

Comment:
I think I see. You specify the arguments to set `local = 1` to avoid global communication. Correct? Perhaps setting `nGrow=1` also would avoid all communication?
The last argument should be a boolean, `true`, to match the argument types of `Redistribute`.

Reply:
Ok, I see you changed to a bool. Though `local` should still be an `int`.

Reply:
Yes, that is correct about `local = 1` (I've actually seen from the collision module that using `true` is okay even though it takes an `int`. I think passing `true` makes more sense, as in `local=true`).
I don't really understand what `nGrow` does. Wouldn't that include a 1 cell wide region around the tile, which would therefore include particles not on the current tile?